Detections/VectraAI/VectraDetect-Account-by-Severity.yaml

```diff
@@ -0,0 +1,65 @@
+id: 95dcf6ac-0527-46fc-8a88-f93d7a9fc059
+name: Vectra AI Detect - Suspected Compromised Account
+description: |
+  'Create an incident when an Account is suspected to be compromised. 
+  The higher the severity level is, the more immediate attention it requires as Vectra AI engine is more confident that this is a real threat. 
+  Level of severity are: Low, Medium, High, Critical). Recommended configuration is to trigger an alert for at least High and Critical.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: AIVectraDetect
+    dataTypes:
+      - CommonSecurityLog
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CredentialAccess
+  - Discovery
+  - LateralMovement
+  - Collection
+  - CommandAndControl
+  - Exfiltration
+  - Impact
+relevantTechniques:
+query: |
+  // Edit this variable to only keep the Severity level where an incident needs to be created (Defaults are: "Low", "Medium", "High", "Critical" )   
+  let configured_level = dynamic(["Low", "Medium", "High", "Critical"]);
+  let upn_has_prefix = ":";
+  CommonSecurityLog
+  | where DeviceVendor == "Vectra Networks"
+  | where DeviceProduct == "X Series"
+  | where DeviceEventClassID == "asc"
+  | extend saccount = extract("saccount=(.+?);", 1, AdditionalExtensions)
+  | extend type = iff(saccount matches regex upn_has_prefix, tostring(split(saccount,":")[0]) ,"network" ) 
+  | extend upn = iff(saccount matches regex upn_has_prefix, tostring(split(saccount,":")[1]) , saccount )
+  | project-rename threat_score = FlexNumber1
+  | project-rename certainty_score = FlexNumber2
+  | project-rename vectra_URL = DeviceCustomString4
+  | project-rename detection_name = DeviceEventClassID
+  | project-rename score_decreases = DeviceCustomString3
+  | extend level = case( threat_score <  50 and certainty_score < 50, "Low",
+                         threat_score < 50 and certainty_score >= 50 , "Medium", 
+                         threat_score >= 50 and certainty_score <= 50, "High", 
+                         threat_score >= 50 and certainty_score >= 50, "Critical",
+                         "UNKNOWN")
+  | extend Severity = case( level == "Low", "Low",
+                            level == "Medium", "Medium",
+                            level == "High", "Medium",
+                            level == "Critical", "High",
+                            "UNKNOWN")
+  | where level in (configured_level) 
+  //keep only the event with the highest threat score per Host
+  | summarize arg_max(threat_score, *) by saccount
+  | project saccount, level, Severity, upn, type, threat_score, certainty_score, vectra_URL
```

Comment:
Do we want to map an account entity in the end using extend - something like 
 | extend AccountCustomEntity = saccount

Also wondering, if these logs have a Timestamp that we need to bring in the results as well as extend in the final line of the query.

Reply:
@aprakash13 , can you explain why "extend AccountCustomEntity = saccount" is required if there is already a section for EntityMapping below the query? I am confused by this.

Reply:
We currently have an old and a new entity mapping method. Since we are still in the transition phase, a good idea is to include both. https://github.com/Azure/Azure-Sentinel/wiki/Query-Style-Guide#entitymappings
Thanks.